File: docs/snippets/all/views/status.py
# Use a blueprint to show a StatusView.

import rerun as rr
import rerun.blueprint as rrb

rr.init("rerun_example_status", spawn=True)

rr.set_time("step", sequence=0)
rr.log("door", rr.Status(status="open"))

rr.set_time("step", sequence=1)
rr.log("door", rr.Status(status="closed"))

rr.set_time("step", sequence=2)
rr.log("door", rr.Status(status="open"))

# Create a status view to display the status transitions.
blueprint = rrb.Blueprint(
    rrb.StatusView(
        origin="/",
        name="Status Transitions",
    ),
    collapse_panels=True,
)

rr.send_blueprint(blueprint)

File: docs/snippets/all/archetypes/status.rs
//! Log a `Status`

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let rec = rerun::RecordingStreamBuilder::new("rerun_example_status").spawn()?;

    rec.set_time_sequence("step", 0);
    rec.log("door", &rerun::Status::new().with_status("open"))?;

    rec.set_time_sequence("step", 1);
    rec.log("door", &rerun::Status::new().with_status("closed"))?;

    rec.set_time_sequence("step", 2);
    rec.log("door", &rerun::Status::new().with_status("open"))?;

    Ok(())
}
